lib/ts/recipe/recipeRouter/index.tsx

```diff
@@ -21,10 +23,33 @@ export abstract class RecipeRouter {
             return undefined;
         }
 
-        const component = routeComponents.find((c) => c.matches());
-        if (component !== undefined) {
-            return component;
+        const dynamicLoginMethods = Multitenancy.getInstanceOrThrow().getLoadedDynamicLoginMethods();
+        const components = routeComponents.filter((c) => c.matches());
+        if (components.length === 1) {
+            return components[0];
         }
+        const possiblyEnabledRecipes =
+            dynamicLoginMethods !== undefined
+                ? {
+                      thirdpartyemailpassword: {
+                          enabled:
+                              dynamicLoginMethods["thirdparty"].enabled && dynamicLoginMethods["emailpassword"].enabled,
+                      },
+                      thirdpartypasswordless: {
+                          enabled:
+                              dynamicLoginMethods["thirdparty"].enabled && dynamicLoginMethods["passwordless"].enabled,
+                      },
+                      ...dynamicLoginMethods,
+                  }
+                : {};
+
+        for (const id in possiblyEnabledRecipes) {
+            const matching = routeComponents.find((c) => c.recipeID === id);
+            if (matching !== undefined) {
+                return matching;
+            }
+        }
```

Comment:
I don't think this works at all. Did you test this? 
- It always selects `thirdpartyemailpassword` even if it's disabled by the core
- when looping through `possiblyEnabledRecipes` you don't check if it's actually enabled or not.
Further issues:
- combination recipes should be enabled if at least 1 sub-recipe is enabled, not if all of them.
- it should prefer an exact matches if possible, not use the first available.
- does this work without dynamic login methods? the old logic was removed and not replaced with anything.

Reply:
if both `thirdparty` and `emailpassword` has `{enabled: false}`, it'll still select `thirdpartyemailpassword`.

if we have only `emailpassword` enabled, we should:
1. render `emailpassword` if it's initialized
2. render `thirdpartyemailpassword`
3. else throw